vignettes/irecist.Rmd

```diff
@@ -0,0 +1,914 @@
+)
+```
+
+## Define Events
+
+The building blocks for the events that contribute to deriving common endpoints
+like what constitutes a responder, or a Best Overall Response of complete
+response (CR), ... are predefined in admiralonco for RECIST 1.1 (see [Pre-Defined
+Response Event Objects](../reference/event_objects.html)). New Response Event Objects 
+are needed for iRECIST and any study-specific needs. 
+
+```{r}
+icpd_y <- event_joined(
+  description = paste(
+    "Define confirmed progressive disease (iCPD) as",
+    "iUPD followed by iCPD with only other iUPD and NE responses in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond_upper = AVALC.join == "iCPD",
+  condition = AVALC == "iUPD" &
+    all(AVALC.join %in% c("iCPD", "iUPD", "NE")),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+iupd_y <- event_joined(
+  description = paste(
+    "Define unconfirmed progressive disease (iUPD) as",
+    "iUPD followed only by other iUPD or NE responses"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "all",
+  condition = ADT <= ADT.join & AVALC == "iUPD" & all(AVALC.join %in% c("iUPD", "NE")),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+no_data_n <- event(
+  description = "Define no response for all patients in adsl (should be used as last event)",
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(AVALC = "N"),
+  keep_source_vars = adsl_vars
+)
+
+no_data_missing <- event(
+  description = paste(
+    "Define missing response (MISSING) for all patients in adsl (should be used",
+    "as last event)"
+  ),
+  dataset_name = "adsl",
+  condition = TRUE,
+  set_values_to = exprs(AVALC = "MISSING"),
+  keep_source_vars = adsl_vars
+)
+```
+
+### Handling Different Input Response Values {#different_resp_vals}
+If `RS` contains other response values than the iRECIST responses, the `event()`
+and `event_joined()` can be adjusted to cover this scenario. For example, if
+RECIST responses (`"CR"`, `"PR"`, `"SD"`, ...) are collected up to first PD and
+iRECIST responses (`"iCR"`, `"iPR"`, `"iSD"`, ...) thereafter, the `event()`
+object defining unconfirmed response can be adjusted in the following way.
+```
+irsp_y <- event(
+  description = "Define CR, iCR, PR, or iPR as (unconfirmed) response",
+  dataset_name = "ovr",
+  condition = AVALC %in% c("CR", "iCR", "PR", "iPR"),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+```
+
+## Derive Confirmed and Unconfirmed Progressive Disease Parameter {#pd}
+
+Now that we have the input records prepared above with any
+company-specific requirements, we can start to derive new parameter
+records. For the parameter derivations, all values except those
+overwritten by `set_values_to` argument are kept from the earliest
+occurring input record fulfilling the required criteria.
+
+When an iCPD occurs, the date of progression would be the same as 
+RECIST 1.1 date (i.e., first `iUPD` date in that block/bar). 
```

Comment:
Feedback from Uwe:
> Not sure if this is always correct: Consider for example SD, SD, iUPD, PR, PR, iUPD,iCPD. RECIST PD date would be the date of the _first_ iUPD while the iRECIST PD date would be the date of the _second_ iUPD.

Reply:
I agree, bad wording, maybe better "When an iCPD occurs, the date of progression would be the first `iUPD` date in that block/bar)".